### PR TITLE
Fix unexpected context canceled when migrating repository

### LINKED ISSUE #27363: No page redirection is performed at the completion of the migration repository
### Description

I have located the problem:  when migration is compend, the `end_time` in the database is not updated.
Maybe  regression of #27089
```
2023/09/30 17:15:29 ...ices/task/migrate.go:58:func1() [E] FinishMigrateTask[31] by DoerID[1] to RepoID[59] for OwnerID[1] failed: context canceled
2023/09/30 17:15:29 ...ices/task/migrate.go:61:func1() [E] runMigrateTask[31] by DoerID[1] to RepoID[59] for OwnerID[1] failed: context canceled
2023/09/30 17:15:29 ...ices/task/migrate.go:68:func1() [E] Task UpdateCols failed: context canceled
2023/09/30 17:15:29 ...ervices/task/task.go:53:handler() [E] Run task failed: context canceled
2023/09/30 17:15:29 ...eb/routing/logger.go:102:func1() [I] router: completed GET /user/task/31 for [::1]:55198, 200 OK in 16.9ms @ user/task.go:16(user.TaskStatus)
```

https://github.com/go-gitea/gitea/assets/50507092/0fd68496-0ce6-4afa-a894-7bb73116e80c

(O9W@2ZHSUI3OFH {@1$HD1

### Gitea Version

try.gitea.io

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

try.gitea.io

### Database

None

### PR DESCRIPTION
**Pull request by CaiCandong**
Fix #27363

Regression of #27089

### CODE REVIEW

File: services/task/migrate.go [hunk 1]
```diff
@@ -49,9 +49,9 @@ func runMigrateTask(ctx context.Context, t *admin_model.Task) (err error) {
 		}
 
 		if err == nil {
-			err = admin_model.FinishMigrateTask(ctx, t)
+			err = admin_model.FinishMigrateTask(db.DefaultContext, t)
```
Comment (delvh) [inline services/task/migrate.go:52]: Hmm…
This does not sound like a good idea.
I mean, I understand that `ctx` is probably canceled after something like 5 seconds, but still, there must be another way of dealing with it…

Comment (CaiCandong) [reply services/task/migrate.go:52]: I know, but I don't have a good idea now.  
This is more like a revert of #27089

**Comment by JakobDev** [comment]
You should add a comment that says the default context is needed and the normal context can't be used

File: services/task/migrate.go (hunk 1 above)
Comment (lunny) [reply services/task/migrate.go:52]: Because ctx is canceled here, so we cannot reuse it.

Comment (lunny) [reply services/task/migrate.go:52]: > Hmm… This does not sound like a good idea. I mean, I understand that `ctx` is probably canceled after something like 5 seconds, but still, there must be another way of dealing with it…

We can find it in another PR but at least fix the bug in this PR.

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Review by puni9869 — APPROVED** [review]

**Pull request merged**